### ISSUE TITLE
ClaRa: replaceHomotopyWithSimplified failing

### ISSUE THREAD

**Issue by AnHeuermann** [OpenModelica/OpenModelica#9043]
### Description

The following models

  - [ClaRa.Components.Control.PredictorModels_3508.Check.TestPredictor](https://libraries.openmodelica.org/branches/master/ClaRa_dev/files/ClaRa_dev_ClaRa.Components.Control.PredictorModels_3508.Check.TestPredictor.err)
  - [ClaRa.Components.VolumesValvesFittings.Pipes.Check.FlowReversal.Test_Pipe_L4_Advanced](https://libraries.openmodelica.org/branches/master/ClaRa_dev/files/ClaRa_dev_ClaRa.Components.VolumesValvesFittings.Pipes.Check.FlowReversal.Test_Pipe_L4_Advanced.err)
  - [ClaRa.Components.VolumesValvesFittings.Pipes.Check.OnePhaseFlow.Test_Pipe_L4_Advanced](https://libraries.openmodelica.org/branches/master/ClaRa_dev/files/ClaRa_dev_ClaRa.Components.VolumesValvesFittings.Pipes.Check.OnePhaseFlow.Test_Pipe_L4_Advanced.err)
  - [ClaRa.Components.VolumesValvesFittings.Valves.Check.Test_GasValves](https://libraries.openmodelica.org/branches/master/ClaRa_dev/files/ClaRa_dev_ClaRa.Components.VolumesValvesFittings.Valves.Check.Test_GasValves.err)

are failing in `replaceHomotopyWithSimplified ` with an error message similar to:

```
Error: Internal error IndexReduction.pantelidesIndexReduction failed! System is structurally singular and cannot be handled because the number of unassigned equations is larger than the number of states. Use -d=bltdump to get more information.
Notification: Performance of <failed> postOpt replaceHomotopyWithSimplified (initialization): time 0.00747/1.298, allocations: 3.194 MB / 0.9986 GB, free: 180.6 MB / 0.6668 GB
Error: post-optimization module replaceHomotopyWithSimplified (initialization) failed.
Error: No system for the symbolic initialization was generated
```

### Steps to Reproduce

```
loadFile("/path/to/TILMediaClaRa/TILMedia/package.mo"); getErrorString();
loadFile("/path/to/clara_om/ClaRa/package.mo"); getErrorString();

simulate(ClaRa.Components.Control.PredictorModels_3508.Check.TestPredictor); getErrorString();
```

### Version and OS

 - OpenModelica Version:  OMCompiler v1.20.0-dev.109+g46cb3ba4a6
 - OS: Ubuntu Focal
 - Versions of used Modelica libraries: ClaRa @ d931ea39e5a56704911d2d4fb78d4386e13ec553

**Comment by casella** [comment]
@AnHeuermann, Dymola always tries to initialize models without using homotopy by default, unless you set `Advanced.OnlyUseHomotopyMethod = true`. The rationale is that if it works, it's faster than with homotopy.

I argued with them in the past that it would be much safer to use homotopy by default (as we do in OMC), because if the modeller put homotopy there, he/she probably had good reasons for it to be used. Also, it may happen that homotopy is used to steer the solver onto a certain solution, in case of nonlinear systems that have many, and that solving the system directly may actually land you on the wrong one. These arguments apparently were not strong enough, so Dymola still tries first without homotopy by default. Maybe sometimes homotopy() is perceived as a kind of remedy for the lack of strong nonlinear solvers, though I would disagree with this opinion.

Anyway, that's the reason why there are many Modelica models around that were built with Dymola, that use `homotopy()`, and that are actually invalid at `lambda = 0`, e.g. because parameters that affect the simplified models are given a zero value, or something like that. As long as Dymola is used to simulate them, and the direct solution without homotopy is successful, nobody will ever know, because this part of the code is never executed. Enter OMC, and the issue is revealed. This actually also happened for some models in the MSL, until we discovered the issue with OMC and fixed them a few years ago.

You could try to run the models setting `Advanced.OnlyUseHomotopyMethod = true` in Dymola (I always do that), and see if you also get an error there, and maybe better diagnostics than with OMC. You can also try to setting the simulation flag [-noHomotopyOnFirstTry](https://www.openmodelica.org/doc/OpenModelicaUsersGuide/latest/simulationflags.html#simflag-nohomotopyonfirsttry) in OMC to see what happens if you skip the lambda = 0 case in OMC.

Skipping the homotopy transformation with a flag to run those models is a possible strategy, but it's not very robust - if homotopy is specified in the model, it should be valid and usable, even if it takes a bit more time than skipping it to solve the initialization problem.

**Comment by AnHeuermann** [comment]
> You could try to run the models setting Advanced.OnlyUseHomotopyMethod = true in Dymola (I always do that), and see if you also get an error there, and maybe better diagnostics than with OMC. 

I'm not super sure,  but I think I activated `Advanced.OnlyUseHomotopyMethod = true` and most of the models are simulating in Dymola.
For `ClaRa.Components.VolumesValvesFittings.Valves.Check.Test_GasValves` I get a bunch of warnings and errors:
```
Warning: Failed to solve nonlinear system using Newton solver.
  During initialization at time: 0
  Tag: initialization.nonlinear[2]
  For debugging information enable
  Simulation/Setup/Debug/Nonlinear solver diagnostics/Detailed logging of failed nonlinear solutions.
Error: could not solve simplified initialization for homotopy method.
Error: Integrator failed to start model.
```

The other issues could also be because of #9036 or another issue with wrong variables.

**Comment by phannebohm** [comment]
Fixed in #9275.

The issue was the initial system with lambda=0. Wie didn't balance it individually, but in fact the actual and simplified expressions may lead to different structures, i.e. causalities.

All of the above models compile now, but some fail during initialization. Perhaps we're still missing something or the lambda0 systems are I'll posed.
